nemo/collections/tts/helpers/helpers.py

```diff
@@ -91,15 +91,16 @@ def get_batch_size(train_dataloader):
 
 
 def get_num_workers(trainer):
-    if trainer.accelerator is None:
+    if trainer._distrib_type is None:
```

Comment:
There's no public API for these anymore ? Can you make sure to use public API if possible ? FYI @blisc

Reply:
Its not clear from their APIs but this is how I found about `distrib_type`. Rest others like `strategy_type` `distributed_backend` aren't always set for various trainer initialization cases

Reply:
That's a PL bug then if it's not set correctly always. We should reach out to them about this, using private variable for a critical part just means it will silently break.

Reply:
This is weird to me. Could you also confirm the same?

There are some accelerator variables assigned to trainer properties, but I couldn't access them. 

```
from pytorch_lightning import Trainer
trainer=Trainer(devices=1,accelerator='gpu',strategy='dp')
print(trainer.num_gpus)# -> works
print(trainer.strategy) # doesnt work
```

Reply:
@blisc my bad, I was viewing the master code base but had 1.5.9 running in the env. 

So currently this is in transition and I see no public api suggesting which property to use. 
I found `trainer.training_type_plugin.distributed_backend.value` would help us transition to 1.6 without any issues but that going to have a deprecated note from 1.6 and would be removed in 1.8 (as seen from master code). I can adobt this put a #TODO note so we can come back to this later once 1.6 is released and we make it a min version in our requirements

Reply:
I support that for now - add a to-do to all places where we need such values from trainer for easy find later, and let's use this deprecated arg for now until Nemo 1.7 is release. We need to discuss with them to have a replacement for critical arguments.

Reply:
Why do we need `strategy`? Can't we just return `devices * num_nodes` or something like that?

Reply:
Unfortunetly this fails for some cases, when user passes `devices=-1` or `devices=[0,1..]` in this cases num_gpus would be good option

Reply:
Oh no, -1 is used across all the Slurm scripts ..

Reply:
Ok, let's rip out this function. It is only used in univnet and hifigan where it is used in computing the max_steps for getting the learning rate. Don't we have some way of doing this in nemo core? Can we just replace this function in TTS with that one?
@titu1994 Do you remember where or if we have such a function?
@Oktai15 Can you confirm that this replace once found is appropriate here?

Reply:
nemo core (modelPT.py) and tts helper (helpers.py) use same way to get num workers. I tried the latest PTL properties but nothing working as expected in a reasonable way (`strategy`, `devices*num_nodes`, `training_type_plugin`) . Its better to look at this after PTL 1.6 gets released or any minor version which brings the suitable property to upgrade to PTL 1.6 safely.

Reply:
It's in the setup_optimization() and I believe it's the same or similar to what's being done here. So all of our optimization would break too. @nithinraok have you reached out to PTL team regarding this ? What was their response if any ?

Reply:
> Can you confirm that this replace once found is appropriate here?

@blisc, yes I can confirm that I took this code from modelPT.py